src/v3/plugins/github/edges.js

```diff
@@ -0,0 +1,216 @@
+// @flow
+
+import {
+  type Edge,
+  type EdgeAddressT,
+  type NodeAddressT,
+  EdgeAddress,
+  NodeAddress,
+} from "../../core/graph";
+import * as GithubNode from "./nodes";
+
+export opaque type RawAddress: EdgeAddressT = EdgeAddressT;
+
+// This particular union of types is used frequently. It seems
+// reasonable that it semantically represents some kind of "post" that
+// has a parent, was authored, and can be referred to. But if one of
+// those axes of symmetry breaks down, then we shouldn't feel hesitant
+// to inline this type in the appropriate places.
+export type PostAddress =
+  | GithubNode.CommentAddress
+  | GithubNode.IssueAddress
+  | GithubNode.PullAddress
+  | GithubNode.ReviewAddress;
+
+export type AuthorsAddress = {|
+  +type: "AUTHORS",
+  +author: GithubNode.UserlikeAddress,
+  +content: PostAddress,
+|};
+export type MergedAsAddress = {|
+  +type: "MERGED_AS",
+  +pull: GithubNode.PullAddress,
+|};
+export type HasParentAddress = {|
+  +type: "HAS_PARENT",
+  +child: PostAddress,
+|};
+export type ReferencesAddress = {|
+  +type: "REFERENCES",
+  +referrer: PostAddress,
+  +referent: PostAddress,
+|};
+
+export type StructuredAddress =
+  | AuthorsAddress
+  | MergedAsAddress
+  | HasParentAddress
+  | ReferencesAddress;
+
+export const createEdge = Object.freeze({
+  authors: (
+    author: GithubNode.UserlikeAddress,
+    content: PostAddress
+  ): Edge => ({
+    address: toRaw({type: "AUTHORS", author, content}),
+    src: GithubNode.toRaw(author),
+    dst: GithubNode.toRaw(content),
+  }),
+  mergedAs: (
+    pull: GithubNode.PullAddress,
+    commitAddress: NodeAddressT /* TODO: Make this a Git commit node address. */
+  ): Edge => ({
+    address: toRaw({type: "MERGED_AS", pull}),
+    src: GithubNode.toRaw(pull),
+    dst: commitAddress,
+  }),
+  hasParent: (
+    child: PostAddress,
+    parentAddress: GithubNode.RawAddress
+  ): Edge => ({
+    address: toRaw({type: "HAS_PARENT", child}),
+    src: GithubNode.toRaw(child),
+    dst: parentAddress,
+  }),
+  references: (referrer: PostAddress, referent: PostAddress): Edge => ({
+    address: toRaw({type: "REFERENCES", referrer, referent}),
+    src: GithubNode.toRaw(referrer),
+    dst: GithubNode.toRaw(referent),
+  }),
+});
+
+const NODE_PREFIX_LENGTH = NodeAddress.toParts(GithubNode._githubAddress())
+  .length;
+
+const GITHUB_PREFIX = EdgeAddress.fromParts(["sourcecred", "github"]);
```

Comment:
i feel like the parts array should be factored out somewhere, and shared between the node and edge. it would be weird if one could change without the other. doesn't necessarily need to happen in this pr.

Reply:
Eh; neutral. It would be weird, but it wouldn’t be wrong (in that it
wouldn’t break anything). I’ll keep things as they are in this PR,
though I am open to factoring this out.